tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,259 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python source file includes CIFAR10 utils for Keras model.
+
+The utilities in this file are used to build a model with native Keras.
+This module file will be used in Transform and generic Trainer.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import List, Text
+
+import os
+import absl
+import tensorflow as tf
+import tensorflow_transform as tft
+
+from tfx.components.trainer.rewriting import converters
+from tfx.components.trainer.rewriting import rewriter
+from tfx.components.trainer.rewriting import rewriter_factory
+
+from tfx.components.trainer.executor import TrainerFnArgs
+
+# CIFAR10 dataset has 50000 train records, and 10000 test records
+_TRAIN_DATA_SIZE = 50000
+_EVAL_DATA_SIZE = 10000
+_TRAIN_BATCH_SIZE = 64
+_EVAL_BATCH_SIZE = 64
+
+IMAGE_KEY = 'image'
+LABEL_KEY = 'label'
+
+def transformed_name(key):
+  return key + '_xf'
+
+def _gzip_reader_fn(filenames):
+  """Small utility returning a record reader that can read gzip'ed files."""
+  return tf.data.TFRecordDataset(filenames, compression_type='GZIP')
+
+def _get_serve_tf_examples_fn(model):
```

Comment:
I mentioned the reason of excluding transform in serving model in the above conversation. I think TFLite rewriter will not have any problem writing the model with transform. The thing here is that usually in serving, models will take image tensors directly, which is also indicated in the MNIST example. MLKit and MediaPipe also requires models to take image tensors directly.

Reply:
The evaluator should work out-of-the-box with both serialized examples as well as tensor-based inputs. What error do you see?

Reply:
I mean it's not the problem with evaluator, and the evaluator will work in both case as long as we feed it with the correct input, i.e. transformed/not transformed. I think the problem is with the saved TFLite model, which will be used for serving in other frameworks. If I understand the process correctly, if we include the transform layer into the model, the saved TFLite model will need to take encoded image strings as input instead of image tensors, which is not compatible with serving frameworks like MLKit and MediaPipe.